src/DGtal/geometry/volumes/BoundedLatticePolytope.h

```diff
@@ -0,0 +1,878 @@
+     *
+     * @param itB the start of the range of n+1 points defining the simplex.
+     * @param itE past the end the range of n+1 points defining the simplex.
+     *
+     * @return 'true' if [itB,itE) was a valid simplex, otherwise
+     * return 'false' and the polytope is empty.
+     *
+     * @note Use DGtal SimpleMatrix::determinant which is not efficient when the
+     * dimension is high. Does not use Eigen to avoid dependency.
+     */
+    template <typename PointIterator>
+    bool init( PointIterator itB, PointIterator itE );
+    
+    /**
+     * Assignment.
+     * @param other the object to copy.
+     * @return a reference on 'this'.
+     */
+    Self & operator= ( const Self & other ) = default;
+
+    /// @}
+
+    // ----------------------- Accessor services ------------------------------
+  public:
+    /// @name Accessor services
+    /// @{
+    
+    /// @return the domain of the current polytope.
+    const Domain& getDomain() const;
+
+    /// @return the number of half-space constraints.
+    unsigned int nbHalfSpaces() const;
+
+    /// @param i the index of the half-space constraint between 0 and
+    /// `nbHalfSpaces()` (excluded).
+    ///
+    /// @return the normal vector of the \a i-th half space
+    /// constraint (i.e. `A` in constraint `Ax <= b`).
+    const Vector& getA( unsigned int i ) const;
+
+    /// @param i the index of the half-space constraint between 0 and
+    /// `nbHalfSpaces()` (excluded).
+    ///
+    /// @return the offset of the \a i-th half space
+    /// constraint (i.e. `b` in constraint `Ax <= b`).
+    Integer getB( unsigned int i ) const;
+
+    /// @param i the index of the half-space constraint between 0 and
+    /// `nbHalfSpaces()` (excluded).
+    ///
+    /// @return 'true' if the \a i-th half space constraint is of the
+    /// form `Ax <= b`, 'false' if it is of the form `Ax < b`.
+    bool isLarge( unsigned int i ) const;
+
+    /// @return the matrix A in the polytope representation \f$ Ax \le B \f$.
+    const InequalityMatrix& getA() const;
+    
+    /// @return the vector B in the polytope representation \f$ Ax \le B \f$.
+    const InequalityVector& getB() const;
+    /// @return the vector I telling if inequalities are large in the
+    /// polytope representation \f$ Ax \prec B \f$, with \f$ \prec \in
+    /// \{ <, \le \} \f$.
+    const std::vector<bool>& getI() const;
+
+    /// @return 'true' if we can perform exact Minkowksi sums on this
+    /// polytope. This is related to the presence of valid edge
+    /// constraints (n-k cells for k >= 2) in-between face constraints
+    /// (n-1 cells) that change orthants.
+    bool canBeSummed() const;
+    
+    /// @}
+    
+    // ----------------------- Check point services ------------------------------
+  public:
+
+    /// @name Check point services (is inside test)
+    /// @{
+
+    /**
+     * @param p any point of the space.
+     * @return 'true' if and only if \a p is inside this polytope.
+     */
```

Comment:
add note on the complexity (this kind of service methods are critical)

Reply:
BTW are you using the tristan's Halfspace orientation predicates for that purpose?